map/autoware_dynamic_lanelet_provider/src/dynamic_lanelet_provider.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright 2024 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "autoware/dynamic_lanelet_provider/dynamic_lanelet_provider.hpp"
+
+namespace autoware
+{
+namespace dynamic_lanelet_provider
+{
+
+// Define a helper function to get x and y
+template <typename T>
+auto getX(const T & point) -> decltype(point.x)
+{
+  return point.x;
+}
+
+template <typename T>
+auto getY(const T & point) -> decltype(point.y)
+{
+  return point.y;
+}
+
+// Define a helper function to get x() and y()
+template <typename T>
+auto getX(const T & point) -> decltype(point.x())
+{
+  return point.x();
+}
+
+template <typename T>
+auto getY(const T & point) -> decltype(point.y())
+{
+  return point.y();
+}
+
+template <typename T, typename U>
+double norm_xy(const T & p1, const U & p2)
+{
+  double dx = getX(p1) - getX(p2);
+  double dy = getY(p1) - getY(p2);
+  return std::hypot(dx, dy);
+}
+
+template <typename T>
+bool is_inside_region(
+  const double & min_x, const double & max_x, const double & min_y, const double & max_y,
+  const T & point)
+{
+  return min_x <= getX(point) && getX(point) <= max_x && min_y <= getY(point) &&
+         getY(point) <= max_y;
```

Comment:
@StepTurtle  Please check the code correctly
```suggestion
  return min_x <= getX(point) && getX(point) >= max_x && min_y <= getY(point) &&
         getY(point) >= max_y;
```

Reply:
Sorry @YamatoAndo, but I didn't understand the change you suggested. Shouldn't the Pose be greater than the min values and smaller than the max values for it to be correct?

Reply:
sry. I didn't notice that the right and left sides were swapped.
I misunderstood that the code was as follows:
```
  return min_x <= getX(point) &&  max_x <= getX(point) && min_y <= getY(point) &&
         max_y  <= getY(point);
```